src/game.cpp

```diff
@@ -508,8 +500,20 @@ void Game::move(int y, int x, int k, int l) {
     score += targetTile.value;
     targetTile.blocked = true;
 
-    largestTile =
-        largestTile < targetTile.value ? targetTile.value : largestTile;
+    if (largestTile < targetTile.value) {
+      largestTile = targetTile.value;
+
+      /**
+       * Default border length is 4. When largestTileLength exceeds 4
+       * then increase the border and cell width by 1
+      */
+      int largestTileLength = std::to_string(largestTile).length();
+      if (cellWidth > MINIMUM_CELL_WIDTH) {
+        cellWidth = largestTileLength;
+        horizontalBorder.append("─");
```

Comment:
With an anonymous-namespace free function, you can take in a parameter `[length]` and the string returned would be a string of the `[horizontalBorder character]`.